### PR TITLE
add init_app method

### PR DESCRIPTION
**Pull request by shenwpo**
issue:  https://github.com/pycasbin/flask-authz/issues/19

    def __init__(self, app=None, adapter=None, watcher=None):
        ...
        if app is not None:
            self.init_app(app)
    def init_app(app):
        ...
    
    casbin_enforcer = CasbinEnforcer(adapter=adapter)
    
    def create_app(config_class=Config):
        app = Flask(__name__)
        app.config.from_object(config_class)
        casbin_enforcer.init_app(app)
        return app

### CODE REVIEW

File: flask_authz/casbin_enforcer.py [hunk 1]
```diff
@@ -16,18 +16,26 @@ class CasbinEnforcer:
...
 
-    def __init__(self, app, adapter, watcher=None):
+    def __init__(self, app=None, adapter=None, watcher=None):
         """
         Args:
             app (object): Flask App object to get Casbin Model
             adapter (object): Casbin Adapter
         """
-        self.app = app
+        self.app = None
```
Comment (jessecooper) [inline flask_authz/casbin_enforcer.py:25]: To quote the example in the flask docs it looks like `self.app = app` and not None:
```
        self.app = app
        if app is not None:
            self.init_app(app)
```

**Comment by jessecooper** [comment]
@shenwpo can you please add tests around these two instantiation paths just to make sure they do not break in the future?

**Comment by shenwpo** [comment]
> @shenwpo can you please add tests around these two instantiation paths just to make sure they do not break in the future?

Well, I'll try to add it.

File: flask_authz/casbin_enforcer.py (hunk 1 above)
Comment (shenwpo) [reply flask_authz/casbin_enforcer.py:25]: I think we should only make a declaration here. The initialization operation relying on app is moved to init_ app. thanks

Comment (jessecooper) [reply flask_authz/casbin_enforcer.py:25]: This makes having `app=None` kwarg pointless if `self.app=None`.  
I would probably suggest the following change:
```
self.app = app
if self.app is not None:
    self.init_app(self.app)
```
This seems like a more straight forward implementation.

Comment (shenwpo) [reply flask_authz/casbin_enforcer.py]: This seems to be a more popular and cleaner implementation.

    self.app = app
    if app is not None:
        self.init_app(app)

**Comment by jessecooper** [comment]
LGTM

**Comment by hsluoyz** [comment]
@Zxilly plz make a new release.

**Comment by Zxilly** [comment]
@hsluoyz have released on pypi and github

**Comment by hsluoyz** [comment]
Released: https://github.com/pycasbin/flask-authz/releases/tag/v2.2.0

**Pull request merged**